### PR TITLE
00-Prelaunch-LIFT

### PR DESCRIPTION
**Pull request by kalxed**

### CODE REVIEW

File: Polaris/src/states/State.cpp [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
-State::State(FlashChip *flash, StateEstimator *stateEstimator, XbeeProSX *xbee, struct Servos *servos, OpenMV *openMV) : flash(flash), stateEstimator(stateEstimator), xbee(xbee), servos(servos), openMV(openMV){}
+State::State(FlashChip *flash, AttitudeStateEstimator *attitudeStateEstimator, XbeeProSX *xbee, struct Servos *servos, OpenMV *openMV) : flash(flash), attitudeStateEstimator(attitudeStateEstimator), xbee(xbee), servos(servos), openMV(openMV){}
 void State::initialize() {
 	this->startTime = millis();
 	initialize_impl();
 	//xbee->begin();
     BLA::Matrix<10> x_0 = {1,0,0,0,0,0,0,0,0,0};
-    stateEstimator = new StateEstimator(x_0, 0.025); 
+    // stateEstimator = new StateEstimator(x_0, 0.025);
+    // I am unaware if there is an AttitudeStateEstimator equivalent to this that should be used Oct 7 2024
```
Comment (kalxed) [inline Polaris/src/states/State.cpp:10]: @frostydev99 Check

File: Polaris/src/states/State.cpp [hunk 2]
```diff
@@ -17,8 +18,8 @@ void State::loop() {
 	loop_impl();
 	this->lastLoopTime = millis();
 	//Sensor stuff here 
-	this->currentState = stateEstimator->onLoop(sensorData); //THis is sus, check pointers and such (tomorrow)
- 
+	//this->currentState = stateEstimator->onLoop(sensorData); //THis is sus, check pointers and such (tomorrow)
+    //Once again- unsure if there is an AttitudeStateEstimator equivalent which should be used Oct 7 2024
```
Comment (kalxed) [inline Polaris/src/states/State.cpp:22]: @frostydev99 Is this true as well?

File: Polaris/src/states/State.cpp (hunk 1 above)
Comment (frostydev99) [reply Polaris/src/states/State.cpp:10]: hmm this is definitely old initialization, there should be a function generating an initial quaternion from accelerometer readings and I don't see that here. Regardless, the update step is doing that so it's not technically necessary.

File: Polaris/src/states/State.cpp (hunk 2 above)
Comment (frostydev99) [reply Polaris/src/states/State.cpp:22]: The attitude state estimator was constructed in a different way than the other objects that are passed between states IIRC (ideally that should be changed)

That should be a static object only initialized once (that was an issue in the past) and that's why its a pointer. If there is a better way of doing this please let me know!

**Pull request merged**